cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java

```diff
@@ -88,13 +106,82 @@ protected void updateView(CoachMarkDimens<Integer> popupDimens,  CoachMarkDimens
                 horizontalPadding, verticalPadding + lowerGap);
     }
 
+    @Override
+    public void show() {
+        super.show();
+
+        if (mHorizontalTranslation) {
+            animateHorizontalTranslation();
+        }
+    }
+
+    /**
+     * Move the punch hole from left to right of the target view, unless the
+     * width of the target view is smaller than the diameter of the punch hole
+     * in which case the circle will be centered and the animation is pointless.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    private void animateHorizontalTranslation() {
+        if (canHaveHorizontalTranslation()) {
+            final int animationStartX = isRtlConfig()
+                    ? mTargetViewLoc[0] + mTargetView.getWidth() - (int) mRelCircleRadius
+                    : mTargetViewLoc[0] + (int) mRelCircleRadius;
+            final int animationEndX = isRtlConfig()
+                    ? mTargetViewLoc[0] + (int) mRelCircleRadius
+                    : mTargetViewLoc[0] + mTargetView.getWidth() - (int) mRelCircleRadius;
+            final ObjectAnimator animator = ObjectAnimator.ofInt(
+                    mPunchHoleView,
+                    "circleCenterX",
+                    animationStartX,
+                    animationEndX);
+            animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+                @Override
+                public void onAnimationUpdate(ValueAnimator animation) {
+                    mPunchHoleView.invalidate();
+                }
+            });
+            // the animation lasts as long as the coachmark is on screen
+            animator.setDuration(mTimeoutInMs);
+            animator.start();
+        }
+    }
+
+    /**
+     * Check if we can animate the circle from left to right, i.e. if the
+     * width of the target view is bigger than the diameter of the circle
+     * (otherwise there's no space to perform the animation).
+     * @return  whether we can display the animation
+     */
+    private boolean canHaveHorizontalTranslation() {
+        return mTargetView.getWidth() > 2 * mRelCircleRadius;
+    }
+
+    /**
+     * For APIs above JELLY_BEAN_MR1, we can check if the locale of a device is
+     * LTR or RTL. Below that API, we assume it's LTR.
+     * We need this to show the horizontal translation from start to end
+     * (from left to right in LTR and from right to left in LTR), if possible.
+     * @return  whether the device has a RTL locale
+     */
+    private boolean isRtlConfig() {
+        final Configuration config = mAnchor.getResources().getConfiguration();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1
+                && config.getLayoutDirection() == View.LAYOUT_DIRECTION_RTL) {
+            return true;
+        }
+        String a = "";
```

Comment:
Ooops, this was something I added to test (I needed something to change in the library, AS issues). I've already removed it.